### DISCUSSION TITLE
change CURL_MAX_WRITE_SIZE do not affect size in CURLOPT_WRITEFUNCTION callback

### DISCUSSION THREAD

**Discussion by BeyondWang** [curl/curl#14210] [Q&A] [answered]
Hello,

Based on the document, I changed CURL_MAX_WRITE_SIZE to 163840 and recompiled libcurl, hoping to let CURLOPT_WRITEFUNCTION callback pass more data each time. But it has no effect, the data size of CURLOPT_WRITEFUNCTION callback is still the default (16384), do I miss anything?
Note: I added some logs in libcurl source codes, and do see the buffer_size is changed to 163840 (log the "url.c -> set->buffer_size = READBUFFER_SIZE;").
libcurl version: 8.8.0

Thanks in advance.

**Comment by icing** [comment, marked as answer]
libcurl will write the amount of data it received from the server. If that exceeds `CURL_MAX_WRITE_SIZE` it will do several writes. If it is less, it writes what it has.

There is no buffering in libcurl until `CURL_MAX_WRITE_SIZE` amount of data is available.

**Reply by BeyondWang** [reply]
Based on the log, 16384 bytes data is writen in the write calllback each time except the last one which is less than 16384, a coincidence?`16384 -> 16384 -> 16384 -> 16384 -> 12288`
Is it possible to detect the original received data size from the server before wirte callback triggered in libcurl?
